project4/3700crawler

```diff
@@ -106,43 +154,42 @@ class WebCrawler:
         # Send the request
         sock = self.connect(host)
         headers = {
-            'cookie': self.cookie,
-            'host': host,
-            'content-length': str(len(body)),
-            'content-type': 'application/x-www-form-urlencoded'
+            'Host': host,
+            'Content-Length': str(len(body)),
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Cookie': self.cookie
         }
         sock.send(str.encode(make_http('POST', path, headers, body)))
 
-        # Get the response
+        # Return the parsed response
         resp = self.get_response(sock)
-
+        resp = self.parse_response(resp)
         return resp
 
     def login(self):
         """Logs into Fakebook"""
+        # Get the log-in form
         login_form = self.GET(LOGIN_FORM)
-        match = re.search('[Ss]et-[Cc]ookie:\W*csrftoken=(\w+);', login_form)
-        if (not match):
-            print('Couldn\'t extract CSRF token, aborting')
-            print(login_form)
-            exit(1)
-        self.csrf = match.group(1)
-        match = re.search('[Ss]et-[Cc]ookie:\W*sessionid=(\w+);', login_form)
-        if (not match):
-            print('Couldn\'t extract SessionID token, aborting')
-            print(login_form)
-            exit(1)
-        self.session = match.group(1)
+        
+        # Loop until the form has been successfully retrieved
+        while login_form['Status Code'] != str(200):
+            login_form = self.GET(LOGIN_FORM)
 
-        self.set_cookie()
-
-        login_resp = self.POST(LOGIN_FORM, make_formstr({
+        # Log in using the given credentials
+        query = {
             'username': self.username,
             'password': self.password,
             'csrfmiddlewaretoken': self.csrf,
             'submit': 'Log in',
             'next': '/fakebook/'
-        }))
+        }
+        body = format_query(query)
+        login_resp = self.POST(LOGIN_FORM, body)
+
+        # Loop until WebCrawler has successfully logged in
+        while login_resp['Status Code'] != str(302):
```

Comment:
Does this work? the response is always gonna be 302 when you attempt to post to that URL the login form.

Reply:
If by works you mean move out of the loop, then yes. These `while` loops are just to check if we ever happen to get an error code after our GET or POST requests.

Reply:
Idk if any other codes could ever come up, but it's to ensure that we get the `200` and the `302` that we want before moving on.

Reply:
... '!=' != '==', I am tired and misread. I think this is like the above, a decent check but one that can cause unexpected behavior. I think erroring-out is more appropriate

Reply:
Okay, I changed it to exit conditions instead.